### DISCUSSION TITLE
Final Hardware Design

### DISCUSSION THREAD

**Discussion by jimahlstrom** [softerhardware/CWKeyer#40] [Ideas]
I think we should remove the sidetone frequency pot. This will allow the other three pots to be spaced apart a little more. The only correct value for the sidetone frequency is the CW tone set in the SDR software. WinKeyer can set this frequency,  but the user docs say to set it the same as in the transceiver. Then the tone heard during Rx of a CW station exactly on the Tx frequency equals the sidetone. And there is never a reason to change the sidetone frequency during a QSO. Generally a CW operator will pick a pleasing sidetone frequency and then train the ear to recognize that frequency. Signals heard on the band are then recognized as being above or below the Tx frequency.

**Comment by recri** [comment]
WinKeyer restricts the sidetone frequency to 2000, 1333, 1000, 800, 666, 570, 500, 440, or 400 Hz.

**Reply by recri** [reply]
Despite WinKeyer's limited frequency selection, I think Jim is right.  It isn't something that needs a knob.

The Teensy 4.X have VBAT and GND for a 3V3 coin cell for a battery backed real time clock.  Not strictly a hardware design issue, since you can get coin cell holders with leads on Amazon solder and stick the holder anywhere in the case, but should be noted as a very easy optional hardware customization, and the software can support it even without the coin cell.  The clock gets synced to host time by the Teensy Loader when you load new firmware.

**Comment by dl1ycf** [comment]
I have modified piHPSDR such that it receives side tone changes from the keyer and adjusts the offset accordingly. If turning the "side tone frequency" pot is not reported to the SDR software, it is mainly useless (I agree). However the pots can be freely assigned by the software, and I think one Pot (for microphone preamp) is badly missing. I think there is a clear hierarchy of importance:

Most important:   CW speed
Important:            Side tone volume
                             Microphone preamp
Unimportant:       Side tone pitch
                             Master volume

I usually change the RX volume in the SDR program (no need to fiddle around with the Master volume knob). I adjust the side tone
level when changing head-phones (different headphones need different levels to make me happy). The Mic preamp setting is necessary since Mic input levels vary drastically between different microphones.

So I think yes, three knobs are enough, but I would assing them to CW speed, Side tone volume, and Mic amplification (if you use the microphone input). BTW in my home-made prototype there is only one knob: for the CW speed, and I am happy with this.
